### PR TITLE
Implemented components for Navigation bar and Search bar

### LINKED ISSUE #27: [TC][FE] Design + Implement NavBar
Story Points: 2
- [ ] make nav bar a react component

### LINKED ISSUE #36: Make button component font size to have em units

### PR DESCRIPTION
**Pull request by shizuko-akamoto**
Two new components, the navigation bar and the search bar. Feel free to pull and play around with the components. I know we didn't have a chance to review the designs for these components so I'm hoping to get any feedback now.

**Changes**

1. `navBar.tsx`
* Implemented the navigation bar with configurable links, some hover styling.
2. `searchBar.tsx`
* Implemented the search bar with dummy results displaying depending on which search filter is selected. Search filters to be supported are defined in `SearchFilter` enum (Place, Trecipe, User). In the actual search bar though, you won't see the User search filter (thought it's not really necessary until we actually support multi-user).
3. Other change
* Renamed `colors.scss` to `variables.scss` (Sorry! And **let me know if you're expecting merge conflicts from this**, I can wait until your code is in and fix the conflicts on my end)
In `variable.scss`, added some more common family and size variables. Updated the existing scss to use those newly defined variables.
* Addressed issue #36, making button.scss to deal with rem units instead of px.

The Trecipe logo here is using the "Shadows in the Light" font we decided upon:
image

Closes #27, closes #36

### CODE REVIEW

File: src/components/NavBar.tsx [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+    url: string;
+}
+export interface NavigationProps {
+    links: Array<Link>;
+}
+
+export class NavBar extends React.Component<NavigationProps, {}> {
+    render() {
+        return (<nav className="nav-bar">
+                {this.props.links.map((link) => <a className="nav-item" href={link.url}>{link.text}</a>)}
```
Comment (knox153) [inline src/components/NavBar.tsx:15]:
```suggestion
                {this.props.links.map((link) => <a className="nav-item" key={someKey} href={link.url}>{link.text}</a>)}
```
There is a warning on browser console if we dont't add a key here. Maybe we can use {link.text} as the key?

File: src/components/SearchBar.tsx [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+    }
+
+    private renderSearchResults() {
+        const {results} = this.state;
+        if (_.isArray(results) && !_.isEmpty(results)) {
+            return (
+                <div className="results-container">
+                    <ul className="results-list">
+                        {results.map((result) => (
+                            <li className="results-entry"><a href="#">{result}</a></li>)
```
Comment (knox153) [inline src/components/SearchBar.tsx:78]:
```suggestion
                            <li key={someKey} className="results-entry"><a href="#">{result}</a></li>)
```
Same thing here

File: src/components/NavBar.tsx (hunk 1 above)
Comment (shizuko-akamoto) [reply src/components/NavBar.tsx:15]: Ah good catch. Thanks, will add that in.

File: src/stylesheets/searchBar.scss [hunk 3]
```diff
@@ -0,0 +1,156 @@
...
+    font-size: $input-font-size;
+    font-family: $default-font-family;
+    background: none;
+    color: $mid-gray;
+    border: none;
+    outline: none;
+    padding: 0 6px;
+  }
+
+  .search-button {
```
Comment (tianyan-zhu) [inline src/stylesheets/searchBar.scss:30]: Can we make the **search button icon** same(similar) size as the other 2 icons in the search bar?
image

File: src/components/Button.tsx [hunk 4]
```diff
@@ -33,7 +33,7 @@ export class Button extends Component<ButtonProps, ButtonState> {
     public static defaultProps: Partial<ButtonProps> = {
         text: 'Button item',
         icon: undefined,
-        fontSize: 14,
+        fontSize: 1,
```
Comment (tianyan-zhu) [inline src/components/Button.tsx:36]: Correct me if I am wrong. I think 1 rem = 16px so if the previous fontSize was 14, I think we should use 0.875 here.

Comment (shizuko-akamoto) [reply src/components/Button.tsx:36]: Done!

File: src/stylesheets/searchBar.scss (hunk 3 above)
Comment (shizuko-akamoto) [reply src/stylesheets/searchBar.scss:30]: I specified the font size in css for the search icon now... before I think it was using the browser default? I can't really tell if it's fixed cuz my default search icon size looks the same as other icons LOL. Let me know if it's fixed.

**Pull request merged**